sdk/iot/provisioning/tests/cmocka/test_az_iot_provisioning_client_sas.c

```diff
@@ -22,7 +22,7 @@
 
 #define TEST_REGISTRATION_ID_STR "myRegistrationId"
 #define TEST_ID_SCOPE "0neFEEDC0DE"
-#define TEST_URL_ENCODED_RESOURCE_URI "0neFEEDC0DE%2fregistrations%2fmyRegistrationId"
+#define TEST_URL_ENCODED_RESOURCE_URI "0neFEEDC0DE/registrations/myRegistrationId"
```

Comment:
Registration id cannot contain upper/lower characters. (https://docs.microsoft.com/en-us/azure/iot-dps/concepts-device#registration-id)
From past experience, having upper characters (or not allowed) was the top cause of issues (especially as certificates needed to be generated).
To set a good example, we should have UTs working with correct registration names.